moveit_ros/moveit_servo/src/servo_node.cpp

```diff
@@ -296,42 +298,45 @@ void ServoNode::servoLoop()
 
   while (rclcpp::ok() && !stop_servo_)
   {
-    // Skip processing if servoing is disabled.
-    if (servo_paused_)
-      continue;
+    // Update the parameters.
+    servo_->updateParams();
 
-    next_joint_state = std::nullopt;
-    const CommandType expected_type = servo_->getCommandType();
-
-    if (expected_type == CommandType::JOINT_JOG && new_joint_jog_msg_)
-    {
-      next_joint_state = processJointJogCommand();
-    }
-    else if (expected_type == CommandType::TWIST && new_twist_msg_)
-    {
-      next_joint_state = processTwistCommand();
-    }
-    else if (expected_type == CommandType::POSE && new_pose_msg_)
-    {
-      next_joint_state = processPoseCommand();
-    }
-    else if (new_joint_jog_msg_ || new_twist_msg_ || new_pose_msg_)
+    // Skip processing commands if servoing is disabled.
+    if (!servo_paused_)
     {
-      new_joint_jog_msg_ = new_twist_msg_ = new_pose_msg_ = false;
-      RCLCPP_WARN_STREAM(LOGGER, "Command type has not been set, cannot accept input");
-    }
+      next_joint_state = std::nullopt;
+      const CommandType expected_type = servo_->getCommandType();
 
-    if (next_joint_state && (servo_->getStatus() != StatusCode::INVALID))
-    {
-      if (servo_params_.command_out_type == "trajectory_msgs/JointTrajectory")
+      if (expected_type == CommandType::JOINT_JOG && new_joint_jog_msg_)
+      {
+        next_joint_state = processJointJogCommand();
+      }
+      else if (expected_type == CommandType::TWIST && new_twist_msg_)
+      {
+        next_joint_state = processTwistCommand();
+      }
+      else if (expected_type == CommandType::POSE && new_pose_msg_)
+      {
+        next_joint_state = processPoseCommand();
+      }
+      else if (new_joint_jog_msg_ || new_twist_msg_ || new_pose_msg_)
       {
-        trajectory_publisher_->publish(composeTrajectoryMessage(servo_->getParams(), next_joint_state.value()));
+        new_joint_jog_msg_ = new_twist_msg_ = new_pose_msg_ = false;
+        RCLCPP_WARN_STREAM(LOGGER, "Command type has not been set, cannot accept input");
```

Comment:
Shouldn't the status be set to something invalid here?

Reply:
Yeah, makes sense.